### ISSUE TITLE
Changing column definition is too slow on MySQL with too many tables

### ISSUE THREAD

**Issue by naoigcat** [doctrine/dbal#5695]
### Bug Report

|    Q        |   A
|------------ | ------
| Version     | 3.4.3

#### Summary

Changing a column definition will `SELECT` from `information_schema.key_column_usage` AND `OPEN_FULL_TABLE` applies to all columns, so it takes a long time if there are many tables.

-   ref. https://dev.mysql.com/doc/refman/5.6/en/information-schema-optimization.html
-   ref. https://dev.mysql.com/doc/refman/5.7/en/information-schema-optimization.html

#### Current behaviour

The migration process does not complete even after several hours.

#### How to reproduce

1.  Create many databases and tables on a MySQL 5.6.10 instance (In my environment, the instance has 6,541 databases and 2,569,217 tables).

2.  Set DB_XXX in .env to connect to the instance.

3.  Make a migration file.

    ```php
    public function up(): void
    {
        Schema::table('products', function (Blueprint $table) {
            $table->unsignedBigInteger('id')->change();
        }
    }
    ```

4.  Run the migration.

    ```sh
    php artisan migrate
    ```

#### Expected behaviour

The migration process will complete. (In my environment with doctrine/dbal version 3.3.8, the migration process completed in a few hundred milliseconds).

**Comment by morozov** [comment]
@naoigcat please try`3.4.5`. This may have been fixed in https://github.com/doctrine/dbal/pull/5667.

**Comment by naoigcat** [comment]
@morozov 

> @naoigcat please try`3.4.5`. This may have been fixed in #5667.

Thank you for the information, I can confirm that this has been fixed with `3.4.5`.
